File: app/src/main/java/arunkbabu90/popmovies/database/MovieDao.java
package arunkbabu90.popmovies.database;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface MovieDao
{
    @Query("SELECT * FROM popular_movies")
    LiveData<List<MoviePopular>> loadPopularMovies();

    @Query("SELECT * FROM top_rated_movies")
    LiveData<List<MovieTopRated>> loadTopRatedMovies();

    @Query("SELECT * FROM favourites ORDER BY Timestamp DESC")
    LiveData<List<Favourite>> loadFavMovies();

    @Query("SELECT FavID FROM favourites WHERE FavID = :id")
    int loadFavouritesById(int id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertFavMovie(Favourite movie);

    @Delete
    void deleteFavMovie(Favourite movie);

    @Query("DELETE FROM favourites WHERE FavID = :id")
    void deleteFavMovieById(int id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertPopularMovies(MoviePopular... movies);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTopRatedMovies(MovieTopRated... movies);
}

File: app/src/main/java/arunkbabu90/popmovies/data/database/MovieDao.kt
package arunkbabu90.popmovies.data.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import arunkbabu90.popmovies.data.model.Movie
import io.reactivex.Completable
import io.reactivex.Flowable

@Dao
interface MovieDao {
    @Query("SELECT * FROM Movie")
    fun getAllMovies(): Flowable<List<Movie>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertMovies(vararg movies: Movie): Completable

    @Query("DELETE FROM Movie")
    fun clearAllMovies()
}